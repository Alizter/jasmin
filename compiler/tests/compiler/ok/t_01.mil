param n : u64;

// we can use a variable name that is ignored
fn foo1(x : stack u64) -> stack u64 * reg u64 * reg bool;

// we can also leave out variable names
fn foo2(stack u64, stack u64[n], x,x,x : stack u64) -> stack u64[n] =
  python print_foo;

// we can use a variable name that is ignored
fn foo3(x, y : stack u64, z : reg bool) -> stack u64 {
  w : stack u64;
  w = x;
  w += x;
  if (w = 5) {
    z, x += w;
  }
  return x;
}

// we can use a variable name that is ignored
fn foo4(x : stack u64) {
}

/*
START:CMD
ARG="print[input][types]"
END:CMD
*/
