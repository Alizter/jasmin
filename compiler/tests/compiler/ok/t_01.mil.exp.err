Processed file tests/compiler/ok/t_01.mil
