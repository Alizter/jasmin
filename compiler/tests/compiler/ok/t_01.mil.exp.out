>> input:
param n : u64;

fn foo1(stack u64) -> stack u64 * reg u64 * reg bool;

fn foo10(x.1 : stack u64, y.2 : stack u64, z.3 : reg bool) -> stack u64 {
  w.4 : stack u64
  (w.4 : stack u64) = (x.1 : stack u64);
  (w.4 : stack u64) += (x.1 : stack u64);
  if $(w.4 : stack u64 = 5) {
    (z.3 : reg bool), (x.1 : stack u64) += (w.4 : stack u64);
    (z.3 : reg bool), (x.1 : stack u64) += (y.2 : stack u64);
  }
  return (x.1 : stack u64);
}

fn foo2(stack u64,stack u64[n : invalid],stack u64,stack u64,stack u64) -> stack u64[n : invalid] = python print_foo;

fn foo3(_x.1 : stack u64) {
}

fn foo4(_x.1 : stack u64) {
}

fn foo5(x.1 : stack u64) {
  (x.1 : stack u64) += (x.1 : stack u64);
}

fn foo6(x.1 : stack u64) -> stack u64 {
  return (x.1 : stack u64);
}

fn foo7(x.1 : stack u64) {
  y.2 : stack u64
  (x.1 : stack u64) += (y.2 : stack u64);
}

fn foo8(x.1 : stack u64) -> stack u64 * stack u64 {
  y.2 : stack u64
  return (y.2 : stack u64),(x.1 : stack u64);
}

fn foo9(x.1 : stack u64) -> stack u64 {
  (x.1 : stack u64) += (x.1 : stack u64);
  return (x.1 : stack u64);
}

